src/model/light_source_model.jl

```diff
@@ -91,34 +93,42 @@ function load_prior()
     # due to the greater flexibility of the galaxy model
     #a = [0.28, 0.72]
     a = [0.99, 0.01]
-    r_mean = Array(Float64, Ia)
-    r_var = Array(Float64, Ia)
     k = Array(Float64, D, Ia)
     c_mean = Array(Float64, B - 1, D, Ia)
     c_cov = Array(Float64, B - 1, B - 1, D, Ia)
 
     v05 = VERSION >= v"0.5.0-dev" ? "-v05" : ""
     stars_file = open(joinpath(cfgdir, "stars$D$v05.dat"))
-    r_fit1, k[:, 1], c_mean[:,:,1], c_cov[:,:,:,1] = deserialize(stars_file)
+    _, k[:, 1], c_mean[:,:,1], c_cov[:,:,:,1] = deserialize(stars_file)
     close(stars_file)
 
     gals_file = open(joinpath(cfgdir, "gals$D$v05.dat"))
-    r_fit2, k[:, 2], c_mean[:,:,2], c_cov[:,:,:,2] = deserialize(gals_file)
+    _, k[:, 2], c_mean[:,:,2], c_cov[:,:,:,2] = deserialize(gals_file)
     close(gals_file)
 
-    # These "magic numbers" have been in use for a while.
-    # They were initially gamma parameters, and now they are log normal
-    # parameters.  TODO: Get rid of these and use an empirical prior.
-    # r = [0.47 1.28; 1/0.012 1/0.11] # These were gamma (shape, scale)
-
-    mean_brightness = [0.47 / 0.012, 1.28 / 0.11 ]
-    var_brightness = [0.47 / (0.012 ^ 2), 1.28 / (0.11 ^ 2) ]
-
-    # The prior contains parameters of a lognormal distribution with
-    # the desired means.
-    r_var = log.(var_brightness ./ (mean_brightness .^ 2) + 1)
-    r_mean = log.(mean_brightness) - 0.5 * r_var
-    PriorParams(a, r_mean, r_var, k, c_mean, c_cov)
+    # log normal parameters for the r-band brightness prior.
+    # these were fit by maximum likelihood to the output of primary
+    # on one field.
+    r_μ = Float64[1.685292911417265, 1.352219729535452]
+    r_σ² = Float64[1.909666346677381^2, 1.160302342323597^2]
+
+    # If I remove this next statement, compile time for
+    # benchmark_infer.jl jumps from 13 seconds to 300 seconds!
+    # Really. It's crazy!
+    #log(42.)
+
+    # Compile time is still over 20x as long if `log(42.)`
+    # is replaced with `exp(2.)`. Insane! Julia 0.5.0 mac os x.
+    # Update: happens on Linux with 0.5.0 binaries too.
+    # exp(2.)
```

Comment:
It's very consistent. If you change `log(42.)` to `exp(2.0)`, the programs takes 5 minutes to compile rather than 13 seconds, even though neither computation is used anywhere in the program.